packages/react/src/components/TreeView/TreeView.tsx

```diff
@@ -202,17 +272,25 @@ export default function TreeView({
   useEffect(() => {
     treeWalker.current =
       treeWalker.current ??
-      document.createTreeWalker(treeRootRef?.current, NodeFilter.SHOW_ELEMENT, {
-        acceptNode: function (node) {
-          if (node.classList.contains(`${prefix}--tree-node--disabled`)) {
-            return NodeFilter.FILTER_REJECT;
-          }
-          if (node.matches(`li.${prefix}--tree-node`)) {
-            return NodeFilter.FILTER_ACCEPT;
-          }
-          return NodeFilter.FILTER_SKIP;
-        },
-      });
+      document.createTreeWalker(
+        treeRootRef?.current as unknown as Node,
+        NodeFilter.SHOW_ELEMENT,
+        {
+          acceptNode: function (node) {
+            if (
+              (node as HTMLElement).classList.contains(
+                `${prefix}--tree-node--disabled`
+              )
+            ) {
+              return NodeFilter.FILTER_REJECT;
+            }
+            if ((node as HTMLElement).matches(`li.${prefix}--tree-node`)) {
+              return NodeFilter.FILTER_ACCEPT;
+            }
+            return NodeFilter.FILTER_SKIP;
+          },
```

Comment:
Maybe it would be more clean to do an early return if the node is not an Element? 

```ts
if (!(node instanceof Element)) { return NodeFilter.FILTER_SKIP; }
```

We are hiding a potential `Cannot access "contains" on undefined` error here by not verifying that the node is an element. I just opted to keep the current implementation behaviors and only add types for now.

Reply:
Thanks for the suggestion. @imp-dance, could you please make this change?

Reply:
@Gururajj77 Done ✅